src/main/java/net/minestom/server/ServerStarter.java

```diff
@@ -0,0 +1,85 @@
+package net.minestom.server;
+
+import lombok.RequiredArgsConstructor;
+import net.minestom.server.exception.ExceptionHandlerProvider;
+import net.minestom.server.monitoring.BenchmarkManagerProvider;
+import net.minestom.server.network.ConnectionManagerProvider;
+import net.minestom.server.network.socket.ServerProvider;
+import net.minestom.server.thread.ChunkDispatcherProvider;
+import net.minestom.server.thread.TickSchedulerThread;
+import net.minestom.server.timer.SchedulerManagerProvider;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@RequiredArgsConstructor
+public class ServerStarter {
```

Comment:
I'm a little unclear on the need for this class

Reply:
This class is needed to start the server (now renamed to ServerProcess). Since MinecraftServer is optional for running Minestom, there should be no logic in it, so the launch logic is in a separate class